packages/material-ui/src/Alert/Alert.js

```diff
@@ -1,147 +1,213 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
+import { deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import { useThemeVariants } from '@material-ui/styles';
-import withStyles from '../styles/withStyles';
+import experimentalStyled from '../styles/experimentalStyled';
 import { darken, lighten } from '../styles/colorManipulator';
 import capitalize from '../utils/capitalize';
 import Paper from '../Paper';
+import { getAlertUtilityClass } from './alertClasses';
 import IconButton from '../IconButton';
 import SuccessOutlinedIcon from '../internal/svg-icons/SuccessOutlined';
 import ReportProblemOutlinedIcon from '../internal/svg-icons/ReportProblemOutlined';
 import ErrorOutlineIcon from '../internal/svg-icons/ErrorOutline';
 import InfoOutlinedIcon from '../internal/svg-icons/InfoOutlined';
 import CloseIcon from '../internal/svg-icons/Close';
 
-export const styles = (theme) => {
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...styles[`${styleProps.variant}${capitalize(styleProps.color || styleProps.severity)}`],
```

Comment:
```suggestion
    ...styles[styleProps.variant],
    ...styles[`${styleProps.variant}${capitalize(styleProps.color || styleProps.severity)}`],
    [`& .${avatarClasses.icon}`]: styles.icon,
    [`& .${avatarClasses.message}`]: styles.message,
    [`& .${avatarClasses.action}`]: styles.action,
```

In order to be sure that you are not missing anything here, you should take a look on what was previously added on the components' `className` in the rendered tree. Previosly we had:

```
      className={clssx(
        classes.root,
        classes[variant],	
        classes[`${variant}${capitalize(color || severity)}`],	
        themeVariantsClassees, // we ignore this, it will automatically be added by the `experimentalStyled` utility
        className, // this is just classname, it has nothing to do with overrides
      )}
```

In addition to this, when there are slots inside the component, like the `icon`, `message` and `action`, you need to add the overrides for these two. You can take a look on the Button for example - https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/Button/Button.js#L24

More generally, the overrides resolver are basically telling the engine how to map the `theme.components.MuiAlert.styleOverrides`. I will see if we need to update the migration guide to make this more clear :)

Reply:
You can probably take a look on the last part of this section of the migration guide and let me know if things are more clear - https://gist.github.com/mnajdova/b0562785b7cf1b9d9aeae382d578594b#experimentalstyled-params